relations/src/merkle_tree/relation.rs

```diff
@@ -27,56 +34,99 @@ pub type SimplePathVar = PathVar<MerkleConfig, LeafHashGadget, TwoToOneHashGadge
 ///
 /// `MerkleTreeRelation` represents a membership proof for a single leaf in a tree on n leaves.
 #[derive(Clone)]
-pub struct MerkleTreeRelation {
+pub struct MerkleTreeRelation<S: State> {
     /// Private witness.
-    pub authentication_path: SimplePath,
+    pub authentication_path: Option<SimplePath>,
```

Comment:
Is it the same as merkle path/merkle proof? It's just I've never seen this term being used.

Reply:
renamed